src/site/content/en/blog/image-support-for-async-clipboard/index.md

```diff
@@ -0,0 +1,317 @@
+
+* The `clipboard-write` permission is granted automatically to pages when they
+  are in the active tab.
+* The `clipboard-read` permission must be requested, which you can do by
+  trying to read data from the clipboard.
+
+```js
+const queryOpts = { name: 'clipboard-read' };
+const permissionStatus = await navigator.permissions.query(queryOpts);
+// Will be 'granted', 'denied' or 'prompt':
+console.log(permissionStatus.state);
+
+// Listen for changes to the permission state
+permissionStatus.onchange = () => {
+  console.log(permissionStatus.state);
+};
+```
+
+## Images in the Asynchronous Clipboard API {: #images }
+
+### Copy: writing an image to the clipboard {: #copy-image }
+
+The new `navigator.clipboard.write()` method can be used for copying images
+to the clipboard. Like [`writeText()`](#copy-text), it is asynchronous, and
+Promise-based. Actually, `writeText()` is just a convenience method for the
+generic `write()` method.
+
+To write an image to the clipboard, you need the image as a [`Blob`][blob].
+One way to do this is by requesting the image from an server by calling
+`fetch()` (or `XMLHTTPReuest()`). The `response` object returned by `fetch()`
+has a [`blob()` method][blob-method] and `XMLHTTPRequest()` let's you set
+`"blob"` as the [`responseType`][blob-response-type].
+
+Calling the server may not be desireable or possible for a variety of reasons.
+Fortunately, you can also write the image to a canvas, then call
+[`HTMLCanvasElement.toBlob()`][to-blob].
+
+Next, pass an array of `ClipboardItem` objects as a parameter to the `write()`
+method. Currently you can only pass one image at a time, but we plan to add
+support for multiple images in the future.
+
+The `ClipboardItem` takes an object with the MIME type of the image as the key,
+and the actual blob as the value. The sample code below shows a future-proof way
+to do this by using the [`Object.defineProperty()`][object-define-prop] method.
+The MIME used as the key is retrieved from `blob.type`. This approach ensures
+that your code will be ready for future image types as well as other MIME types
+that may be supported in the future.
+
+```js
+try {
+  const imgURL = '/images/generic/file.png';
+  const data = await fetch(imgURL);
+  const blob = await data.blob();
+  await navigator.clipboard.write([
+    new ClipboardItem(Object.defineProperty({}, blob.type, {
+      value: blob,
+      enumerable: true
+    }))
+  ]);
+  console.log('Image copied.');
+} catch(e) {
+  console.error(e, e.message);
+}
+```
+
+### Paste: reading an image from the clipboard {: #paste-image }
+
+The `navigator.clipboard.read()` method, which reads data from the clipboard, is
+also asynchronous, and Promise-based.
+
+To read an image from the clipboard, obtain a list of `ClipboardItem` objects,
+then iterate over them. Since everything is asynchronous, use the
+[`for ... of`][for-of] iterator, since it handles async/await code nicely.
```

Comment:
```suggestion
[`for...of`][for-of] iterator, since it handles async/await code nicely.
```
Removing the spaces to match the syntax in the MDN doc that's linked here